src/matrix/internal/ProjectedVector.ts

```diff
@@ -34,6 +35,7 @@ export default class ProjectedVector<T, D extends IValueTypeDesc, M, MD extends
       creator: m.desc.creator,
       ts: m.desc.ts
     };
+    this.parentValueType = (<any>this.m).root.valuetype.range; //Hack to access root range for matrix to vector conversion for rendering barplot.
```

Comment:
`this.valuetype` only give the range from the matrix view created from its root matrix. While `this.parentValueType` gives the range from the parent matrix